documentation/public/user-guide/08-examples/01-list-functions-in-contract/examples/list-functions-in-contract.mts

```diff
@@ -0,0 +1,29 @@
+import assert from "node:assert";
+import { Query } from "@nomicfoundation/slang/cst";
+import { ContractDefinition, FunctionDefinition } from "@nomicfoundation/slang/ast";
+import { CompilationUnit } from "@nomicfoundation/slang/compilation";
+
+export function listFunctionsInContract(unit: CompilationUnit, contractName: string): FunctionDefinition[] | undefined {
```

Comment:
Should we return an array when the contract is not found, instead of returning `undefined`?
If it was an error/unexpected case, I suggest `throw`ing an `Error` with the exact error message, as it is more idiomatic, and the user won't have to assert the type at the callsite in the happy path..